src/java.desktop/share/classes/javax/swing/plaf/basic/BasicScrollBarUI.java

```diff
@@ -1202,6 +1218,35 @@ public boolean getSupportsAbsolutePositioning() {
         return supportsAbsolutePositioning;
     }
 
+    /**
+     * A listener to listen for keyboard focus changes.
+     */
+    protected class KeyboardFocusListener extends MouseAdapter implements PropertyChangeListener {
```

Comment:
Why do you extends "MouseAdapter" ?
Is it really needed ?

Reply:
Looks like an artifact from my other implementations. This isn't required anymore, thanks. Removed.